cortex-r-rt/src/lib.rs

```diff
@@ -446,6 +449,25 @@ core::arch::global_asm!(
     .global _default_start
     .type _default_start, %function
     _default_start:
+        // only allow cpu0 through for initialization
+        // Read MPIDR
+        mrc	    p15,0,r1,c0,c0,5
+        // Extract CPU ID bits by reading affinity level 0.
+        // For single-core systems, this should always be 0.
+        and	    r1, r1, #0xFF
```

Comment:
Maybe we should check the bottom 16 bits are zero (affinity level 0 and affinity level 1)?